### DISCUSSION TITLE
dell-self-test and uefi-self-test fail when building master

### DISCUSSION THREAD

**Discussion by Sean-StarLabs** [fwupd/fwupd#3518] [General]
As the title suggests, probably something I am doing as I couldn't see a change that would affect it though hence raising it here.

Would you be able to point me in the right direction?

Build log:
https://launchpadlibrarian.net/549523495/buildlog_ubuntu-focal-amd64.fwupd_1.6.2+slx+183+rg0920c440_BUILDING.txt.gz

**Comment by hughsie** [comment]
I think you probably need to do something like `export CACHE_DIRECTORY=/tmp` so that you get a writeable cachedir for the tests.

**Reply by Sean-StarLabs** [reply]
That does the trick, thank you :)

**Reply by superm1** [reply]
Is this needed in contrib packaging directory somewhere?

**Reply by Sean-StarLabs** [reply]
Yup - contrib/debian/rules

**Reply by superm1** [reply]
I wonder why this isn't failing our CI though.

**Reply by Sean-StarLabs** [reply]
Not sure - can't blame Ubuntu/Launchpad as the same happened when the Manjaro tried to package it up

**Reply by hughsie** [reply]
My RHEL build also failed, but Fedora was unaffected. We should probably fix the tests.

**Reply by Sean-StarLabs** [reply]
`export CACHE_DIRECTORY=/tmp` doesn't do the trick on Manjaro

**Reply by Sean-StarLabs** [reply]
Slightly related, but the combo of latest tags (fwupd 1.6.1 + fwupd-efi 1.1) doesn't work for flashrom and EFI capsules.

Flashrom ends up with `"failed to open device for prepare: flash probe failed: no chip was found"` and EFI Capsule's get stuck a BGRT indefinitely. Probably something you both know about as both are fixed in master.

Some distro's package things up in seconds based on tags, presumably with little testing so the broken versions seem to hang around for a while. So to stop waffling:
* Is it possible to tag a new release fairly soon?
* Can we help with testing in any way?

**Reply by superm1** [reply]
>Flashrom ends up with "failed to open device for prepare: flash probe failed: no chip was found" and EFI Capsule's get stuck a BGRT indefinitely. Probably something you both know about as both are fixed in master.

I wouldn't make that presumption.  Can you please check on master before we go tag a release and end up with these things in another release?

**Reply by Sean-StarLabs** [reply]
I was about to say "I checked this morning" but it appears that was too long ago!

I'll test again...

**Reply by Sean-StarLabs** [reply]
All working correct as of 2ece14d9e34b0089d1386207b460ff1fa9f457e6

**Reply by Sean-StarLabs** [reply]
Self tests are failing again, build log if it helps:
https://launchpadlibrarian.net/556409969/buildlog_ubuntu-focal-amd64.fwupd_1.7.0.107+rg938f4832+f_BUILDING.txt.gz

**Reply by superm1** [reply]
See if https://github.com/fwupd/fwupd/pull/3696 helps

**Reply by Sean-StarLabs** [reply]
Afraid not:
https://launchpadlibrarian.net/556469316/buildlog_ubuntu-focal-amd64.fwupd_1.7.0.110+rgefd945b1+f_BUILDING.txt.gz

**Reply by superm1** [reply]
Try with the new fix and see if that helps now

**Reply by Sean-StarLabs** [reply]
It does indeed - thank you :)
